src/components/FeaturePanel/Climbing/utils/ascents.ts

```diff
@@ -0,0 +1,75 @@
+import { JSONValue } from '../types';
+
+const KEY = 'ascents';
+
+// @TODO generic util?
+export const getLocalStorageItem = (key: string) => {
+  const raw = window?.localStorage.getItem(key);
+  if (raw) {
+    try {
+      const json = JSON.parse(raw);
+
+      return json;
+    } catch (e) {
+      return [];
+    }
+  }
+  return [];
+};
+
+// @TODO generic util?
+export const setLocalStorageItem = (key: string, value: JSONValue) => {
+  window?.localStorage.setItem(key, JSON.stringify(value));
+};
+
+export const onAscentAdd = ({ osmId }) => {
+  if (!osmId) return;
+  const ascents = getLocalStorageItem(KEY);
+  setLocalStorageItem(KEY, [
+    ...ascents,
+    { id: osmId, date: new Date().toISOString() },
+  ]);
+};
+
+export const findAscents = (osmId: string) => {
+  const ascents = getLocalStorageItem(KEY);
+  return ascents?.filter((ascent) => osmId === ascent.id) ?? null;
+};
+
+export const onAscentDelete = ({
+  osmId,
+  index,
+}: {
+  osmId: string;
+  index: number;
+}) => {
+  const ascents = getLocalStorageItem(KEY);
+
+  const newArray = ascents.reduce(
```

Comment:
what about using just ascents.filter()? This looks so complicated 😅 

Take inspiration here: https://github.com/zbycz/osmapp/blob/master/src/components/utils/StarsContext.tsx#L33